scrapy/spidermiddlewares/urllength.py

```diff
@@ -20,7 +21,7 @@ def __init__(self, maxlength):
     @classmethod
     def from_settings(cls, settings):
         maxlength = settings.getint('URLLENGTH_LIMIT')
-        if not maxlength:
+        if maxlength is None:
```

Comment:
The expectation is for it to be 0.

```suggestion
        if not maxlength:
```